record/record.go

```diff
@@ -233,6 +239,48 @@ func (r *Reader) Recover() {
 	return
 }
 
+// SeekRecord seeks in the underlying io.Reader such that calling r.Next returns
+// the record whose first chunk header starts at the given offset in the
+// underlying io.Reader. If there is an unrecovered error, the caller should
+// call Recover before calling SeekRecord. SeekRecord will clear the pending
```

Comment:
"SeekRecord will clear the pending error" doesn't feel right. SeekRecord doesn't clear any errors, Recover does. It also feels awkward to put the text about Recover between "the given offset" above and "if the argument given is not such an offset" below. How about:

// SeekRecord seeks in the underlying io.Reader such that calling r.Next
// returns the record whose first chunk header starts at the given offset in
// the underlying io.Reader. Its behavior is undefined if the argument given is
// not such an offset, as the bytes at that offset may coincidentally appear to
// be a valid header.
//
// SeekRecord will fail and return an error if the Reader previously
// encountered an error, including io.EOF. Such errors can be cleared by
// calling Recover. Calling SeekRecord after Recover will make calling Next
// return the record at the given offset, instead of the record at the next
// good 32KiB block as Recover normally would. Calling SeekRecord before
// Recover has no effect on Recover's semantics other than changing the
// starting point for determining the next good 32KiB block.
//
// The offset is always relative to the start of the underlying io.Reader, so
// negative values will result in an error as per io.Seeker.
//
// It returns ErrNotAnIOSeeker if the underlying io.Reader does not also
// implement io.Seeker.